src/huggingface_hub/repository.py

```diff
@@ -393,10 +428,40 @@ def git_pull(self, rebase: Optional[bool] = False):
         except subprocess.CalledProcessError as exc:
             raise EnvironmentError(exc.stderr)
 
-    def git_add(self, pattern="."):
+    def auto_track_large_files(self, pattern="."):
+        """
+        Automatically track large files with git-lfs
+        """
+        try:
+            p = subprocess.run(
+                ["git", "ls-files", "-mo", pattern],
```

Comment:
in my tests this also lists files that are gitignored

Reply:
not sure if there's a workaround

Reply:
Oh, exceptional catch! Actually, this is somewhat of a workaround:
```py
git ls-files -mo -X .gitignore 
```
But it doesn't look for nested `.gitignore`s.

However I'm not sure if this is actually important, as these files won't get `git add`ed so they won't end up staged?

Reply:
probably not very important (other than increasing the size of the .gitattributes file but i guess that's not a huge deal – unless it gets to > 10MB)

Reply:
If it's important I'm sure I can find a workaround - let me know if it's high priority

Reply:
no I don't think it's a big deal at all